src/redux/services/fileEntry.ts

```diff
@@ -348,13 +351,78 @@ function addFile(absolutePath: string, state: AppState, appConfig: AppConfig) {
   resourcesFromFile.forEach(resource => {
     state.resourceMap[resource.id] = resource;
   });
-
   reprocessResources(
     resourcesFromFile.map(r => r.id),
     state.resourceMap,
     state.fileMap,
     state.resourceRefsProcessingOptions
   );
+
+  const parentFolderAbsPath = path.dirname(absolutePath);
+  let parentFolderEntry: FileEntry | undefined;
+
+  if (parentFolderAbsPath === rootFolderEntry.filePath) {
+    parentFolderEntry = rootFolderEntry;
+  } else {
+    const parentFolderRelPath = parentFolderAbsPath.substring(rootFolderEntry.filePath.length);
+    const parentFolderEntryKey = Object.keys(state.fileMap).find(key => key === parentFolderRelPath);
+    if (parentFolderEntryKey) {
+      parentFolderEntry = state.fileMap[parentFolderEntryKey];
+    }
+  }
+
+  // if this file is the Helm Chart entry file, create a new helm chart and search for exising values files
+  const isHelmChartFile = path.basename(absolutePath) === HELM_CHART_ENTRY_FILE;
+  if (isHelmChartFile) {
+    const helmChart: HelmChart = {
+      id: uuidv4(),
+      filePath: fileEntry.filePath,
+      name: parentFolderAbsPath.trim() !== '' ? path.basename(parentFolderAbsPath) : 'Unnamed Chart',
+      valueFileIds: [],
+    };
+    state.helmChartMap[helmChart.id] = helmChart;
+
+    parentFolderEntry?.children?.forEach(fileName => {
+      if (!parentFolderEntry) {
+        return;
+      }
+      const isValuesFile = micromatch.isMatch(fileName, '*values*.yaml');
+      if (isValuesFile) {
+        const valuesFilePath =
+          parentFolderEntry.filePath === rootFolderEntry.filePath
+            ? `${path.sep}${fileName}`
+            : path.join(parentFolderEntry.filePath, fileName);
+        const helmValuesFile: HelmValuesFile = {
+          id: uuidv4(),
+          filePath: valuesFilePath,
+          name: fileName,
+          isSelected: false,
+          helmChartId: helmChart.id,
+        };
+        helmChart.valueFileIds.push(helmValuesFile.id);
+        state.helmValuesMap[helmValuesFile.id] = helmValuesFile;
+      }
+    });
+  }
+
+  // if this new file is a values file, search for it's helm chart and update it
+  if (micromatch.isMatch(path.basename(absolutePath), '*values*.yaml')) {
+    const helmChart = Object.values(state.helmChartMap).find(chart => {
+      return path.dirname(chart.filePath) === parentFolderEntry?.filePath;
+    });
+    if (helmChart) {
+      const helmValuesFile: HelmValuesFile = {
+        id: uuidv4(),
+        filePath: relativePath,
+        name: path.basename(relativePath),
+        isSelected: false,
+        helmChartId: helmChart.id,
+      };
+      state.helmValuesMap[helmValuesFile.id] = helmValuesFile;
+      helmChart.valueFileIds.push(helmValuesFile.id);
+    }
```

Comment:
as above - can some of this be extracted and shared with processing already in helm.ts !?